airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -307,6 +329,15 @@ private static Runnable readFromSrcAndWriteToDstRunnable(final AirbyteSource sou
 
             messageTracker.acceptFromSource(message);
 
+            if (message.getType() == Type.CONTROL && message.getControl().getType() == AirbyteControlMessage.Type.CONNECTOR_CONFIG) {
+              try {
+                persistConfigHelper.persistSourceConfig(jobId, message.getControl().getConnectorConfig().getConfig());
+              } catch (final ApiException e) {
+                LOGGER.error("Error trying to save updated source config", e);
+                throw new RuntimeException(e);
+              }
+            }
+
```

Comment:
same though here on pulling this out into a private method.

Reply:
👍🏻 done in [51f1976](https://github.com/airbytehq/airbyte/pull/19811/commits/51f19765184523e0ae88d8ffceaaad6e39a772e1)